src/gf/seismosizer.py

```diff
@@ -1666,16 +1685,39 @@ class RectangularSource(DCSource):
         default=1)
 
     def base_key(self):
-        return DCSource.base_key(self) + (
+        return SourceWithDerivedMagnitude.base_key(self) + (
+            self.magnitude,
+            self.slip,
+            self.strike,
+            self.dip,
+            self.rake,
             self.length,
             self.width,
             self.nucleation_x,
             self.nucleation_y,
-            self.velocity,
-            self.slip)
+            self.velocity)
 
-    def discretize_basesource(self, store, target=None):
+    def check_conflicts(self):
+        if self.magnitude is not None and self.slip is not None:
+            raise DerivedMagnitudeError(
+                'magnitude and slip are both defined')
+
+    def get_magnitude(self, store=None, target=None):
+        self.check_conflicts()
+        if self.magnitude is not None:
+            return self.magnitude
+
+        elif self.slip is not None:
+            amplitudes = self._discretize1(store, target)[2]
+            return float(mt.moment_to_magnitude(num.sum(amplitudes)))
+
+        else:
+            return float(mt.moment_to_magnitude(1.0))
+
+    def get_factor(self):
+        return 1.0
 
+    def _discretize1(self, store, target):
```

Comment:
why this 1 in the discretize. The factor is clear ;) !

Reply:
why not only _discretize ? With this one somehow one expects a _discretize2

Reply:
ok.